test/unit/test_jinja.py

```diff
@@ -413,3 +415,61 @@ def test_if_endfor_newlines(self):
 '''
 
 
+native_expected_behaviors = [
+    # strings
+    ('''foo: bar''', 'bar'),
+    ('''foo: "bar"''', 'bar'),
+    ('''foo: "'bar'"''', "'bar'"),
+    ("""foo: '"bar"'""", '"bar"'),
+    # ints
+    ('''foo: 1''', 1),
+    ('''foo: "1"''', 1),
+    ('''foo: "'1'"''', "'1'"),
+    ('''foo: "{{ 1 }}"''', 1),
+    ('''foo: "{{ '1' }}"''', 1),
+    ('''foo: "'{{ 1 }}'"''', "'1'"),
+    ('''foo: "'{{ '1' }}'"''', "'1'"),
+    ('''foo: "{{ 1 | as_text }}"''', '1'),
+    ('''foo: "{{ '1' | as_text }}"''', '1'),
+    # booleans.
+    # Note the discrepancy with true vs True: `true` is recognized by jinja but
+    # not literal_eval, but `True` is recognized by ast.literal_eval.
+    # For extra fun, yaml recognizes both.
+    ('''foo: "{{ true }}"''', True),
+    ('''foo: "{{ 'true' }}"''', 'true'),
+    ('''foo: "'{{ true }}'"''', "'True'"),
+    ('''foo: "{{ true | as_text }}"''', "True"),  # true -> boolean True -> text -> str(True) -> 'True'
+    ('''foo: "{{ 'true' | as_text }}"''', "true"),  # 'true' -> string 'true' -> text -> str('true') -> 'true'
+    ('''foo: "{{ True }}"''', True),
+    ('''foo: "{{ 'True' }}"''', True),
+    ('''foo: "'{{ True }}'"''', "'True'"),
+    ('''foo: "{{ True | as_text }}"''', "True"),  # True -> string 'True' -> text -> str('True') -> 'True'
+    ('''foo: "{{ 'True' | as_text }}"''', "True"),  # 'True' -> string 'True' -> text -> str('True') -> 'True'
+    ('''foo: yes''', True),  # yaml turns 'yes' into a boolean true
+    ('''foo: "yes"''', "yes"),
+    # concatenation
+    ('''foo: "{{ a_int + 100 }}"''', 200),
+    ('''foo: "{{ a_str ~ 100 }}"''', 100100),
+    ('''foo: "{{ a_int ~ 100 }}"''', 100100),
+    ('''foo: "{{ a_str }}{{ a_str }}"''', 100100),
+    ('''foo: "{{ a_int }}{{ a_int }}"''', 100100),
```

Comment:
can you add one more test here for:
```
'''foo: "'{{ a_int }}{{ a_int }}'"'''
```

I would expect to see `"100100"` (as a string) but it's worth 1) double checking and 2) adding the test for it

Reply:
This actually ends up as the string `'100100'` (including quotes) - it's similar to the `'''foo: "'{{ 1 }}'"'''` test case.

Reply:
I am _so_ sorry, I got the example wrong. You're right - I'd expect to see the single quotes in here too! `'100100'` is indeed what this should render out to